### PR TITLE
(CDAP-1195) Basic implementation that uses Hive Predicate Pushdown to se...

### PR DESCRIPTION
**Pull request by chtyim**
...t start/end time when querying stream events.

### CODE REVIEW

File: cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java [hunk 1]
```diff
@@ -24,8 +24,21 @@
 import co.cask.cdap.data2.transaction.stream.StreamConfig;
 import co.cask.cdap.hive.context.ContextManager;
 import co.cask.cdap.proto.Id;
+import com.clearspring.analytics.util.Lists;
```
Comment (albertshau) [inline cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java]: change to guava

File: cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java [hunk 2]
```diff
@@ -94,12 +106,99 @@ public InputSplit createSplit(Path path, Path indexPath, long startTime, long en
...
+    }
+
+    // Stream can only be indexed by timestamp
+    analyzer.clearAllowedColumnNames();
+    analyzer.allowColumnName("ts");
+
+    ExprNodeGenericFuncDesc expr = Utilities.deserializeExpression(serializedExpr);
+
+    List<IndexSearchCondition> conditions = Lists.newArrayList();
+    analyzer.analyzePredicate(expr, conditions);
```
Comment (albertshau) [inline cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java]: javadoc says this only supports "pure conjunctions over binary expressions". What happens when you analyze something that isn't? like "ts < X or ts > Y"?

Comment (albertshau) [inline cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java]: can we add a test case to make sure it doesn't throw and exception and fail the query

**Comment by albertshau** [comment]
one minor comment plus a question.  Have we tried this on multiple Hive versions? We sometimes run into problems on different versions.

File: cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java (hunk 1 above)
Comment (chtyim) [inline cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java]: I should add this class to abandon list in my IDE

File: cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java (hunk 2 above)
Comment (chtyim) [inline cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java]: It doesn't handle "or" case as far as I tested. We can enhance the support later in this class.

Comment (chtyim) [inline cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java]: For expression that the `IndexAnalyzer" not able to match anything, the list will be empty.

**Comment by albertshau** [comment]
lgtm assuming it's been tested on a cluster

**Comment by albertshau** [comment]
:+1:

**Comment by chtyim** [comment]
Build passed and tested on cluster: http://builds.cask.co/browse/CDAP-DUT1241-4

**Pull request merged**